File: springs_setup.py
import sys
import os
from pathlib import Path

def main(argv):
	dir_solver = Path('.') / 'springs_solver'
	dir_solver_obj = dir_solver / 'obj'
	os.system( 'mkdir ' + str(dir_solver_obj) )
	os.system( 'make -C ' + str(dir_solver) + ' clean' )
	os.system( 'make -C ' + str(dir_solver) )

if __name__ == '__main__':
	main(sys.argv[1:])

File: springs_matlab_solver/springs_setup.m
function springs_setup()

dir_solver = fullfile('.','springs_solver') ;
dir_solver_obj = fullfile(dir_solver,'obj') ;
if ~exist(dir_solver_obj)
	mkdir(dir_solver_obj)
end
system(sprintf('make -C %s clean',dir_solver)) ;
system(sprintf('make -C %s',dir_solver)) ;

end